### ISSUE TITLE
Make reactive messaging throttled strategy put messages in DLQ when throttled.unprocessed-record-max-age.ms is reached instead of stopping the connector from consuming.

### ISSUE THREAD

**Issue by javadevmtl** [quarkusio/quarkus#37572]
### Description

Currently the the behaviour is to stop processing/consuming. Which then requires restart of the application, be it manual or through devops health checks automation.

Also what is not clear when the application is restarted if it goes back to that messages offset and it's a true "poison pill" then technically speaking you will never be able to get past that record or when the application is restarted it skips to latest offset then there a chance you miss processing records? (This happened recently to us, where the consumer unregistered we restarted the app and any new record behind the records that got stuck where just skipped because the app started at the "latest" offset)

It would be nice if when this warning happens, that we have the option to DLQ the message and then let us decide wether it goes into a retry mechanism or just stays DLQ.

**Comment by cescoffier** [comment]
Did you configure the DLQ failure strategy?

**Comment by javadevmtl** [comment]
Yes. It's configured it just stops consuming and marks itself unhealthy.

**Comment by cescoffier** [comment]
Is it a deserialization issue or a processing issue?

**Comment by ozangunalp** [comment]
I think this is related to this question: https://stackoverflow.com/questions/77615139/att-what-offset-does-quarkus-kafka-messaging-recover-after-throttled-unprocessed ? 

Is it possible that the consumer offset retention kicks in and cleans the offset commits of your consumer group? By default, the retention should be one week. It is the only thing that comes to my mind when you say that the consumer state is not there.

Otherwise, for the question of this issue, It would need the throttled commit strategy to keep references to messages pending for processing completion. A similar thing you can do is to adjust the `throttled.unprocessed-record-max-age.ms` and put a timeout to your processing – maybe to the HTTP call you are making, so that the processing will timeout before the max-age. Than the failure strategy can kick in and send that message to the DLQ.

**Comment by javadevmtl** [comment]
Hi yes, it is related to that thread. Ok we will look at increasing the retention. So if we get this warning, 100% if we restart the consuming application in a timely manner then it will pick up right at that precise message, effectively making it at least once?

The thing is this feature indicates that it's to try to avoid "poison pill" type message, if the message is truly a poison pill then you will never get past it?

Maybe I'm the only one complaining, I guess, but it truly feels that if the warning is caught/detected then that message should be DLQed as an option and not only when we throw an exception.
